rules/Arguments/ArgumentDefaultValueReplacer.php

```diff
@@ -68,9 +66,9 @@ public function isDefaultValueMatched(?Expr $expr, $value): bool
     }
 
     private function processParams(
-        ClassMethod $classMethod,
-        ReplaceArgumentDefaultValueInterface $replaceArgumentDefaultValue
-    ): ?ClassMethod {
+        object $classMethod,
+        \ReplaceArgumentDefaultValueInterface $replaceArgumentDefaultValue
```

Comment:
it seems may cause invalid replacement:

```diff
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
1 file with changes
===================

1) rules/Arguments/ArgumentDefaultValueReplacer.php:3

    ---------- begin diff ----------
@@ @@

 namespace Rector\Arguments;

+use object;
 use PhpParser\BuilderHelpers;
 use PhpParser\Node;
 use PhpParser\Node\Arg;
@@ @@
         return $this->processArgs($node, $replaceArgumentDefaultValue);
     }

-    /**
-     * @param mixed $value
-     */
-    public function isDefaultValueMatched(?Expr $expr, $value): bool
+    public function isDefaultValueMatched(?Expr $expr, mixed $value): bool
     {
         // allow any values before, also allow param without default value
         if ($value === ReplaceArgumentDefaultValue::ANY_VALUE_BEFORE) {
@@ @@
     }

     private function processParams(
-        ClassMethod $classMethod,
-        ReplaceArgumentDefaultValueInterface $replaceArgumentDefaultValue
-    ): ?ClassMethod {
+        object $classMethod,
+        \ReplaceArgumentDefaultValueInterface $replaceArgumentDefaultValue
+    ): ?object {
         $position = $replaceArgumentDefaultValue->getPosition();

         if (! $this->isDefaultValueMatched(
@@ @@
    ----------- end diff -----------

Applied rules:
 * CountOnNullRector (https://3v4l.org/Bndc9)
 * MixedTypeRector
 * AddMethodCallBasedStrictParamTypeRector (https://github.com/symplify/phpstan-rules/blob/master/docs/rules_overview.md#checktypehintcallertyperule)
 * ReturnTypeDeclarationRector (https://wiki.php.net/rfc/scalar_type_hints_v5)
```

Reply:
The issue seems when combined with `AddMethodCallBasedStrictParamTypeRector`

Reply:
tweak by check virtual node https://github.com/rectorphp/rector-src/pull/2701/commits/39e3969a2a4884cf6bceeee558f36ceef4f4c5d0